symtabAPI/src/parseStab.C

```diff
@@ -220,17 +209,7 @@ std::string Dyninst::SymtabAPI::parseStabString(Module *mod, int linenum, char *
    std::string mangledname = getIdentifier( stabstr, cnt );
 
    currentRawSymbolName = mangledname;
-   char * demangled = P_cplus_demangle( mangledname.c_str(), mod->exec()->isNativeCompiler() );
-   std::string name;
-
-   if ( demangled == NULL ) 
-   {
-      name = mangledname;
-   } 
-   else 
-   {
-      name = demangled;
-   }
+   std::string name = P_cplus_demangle( mangledname );
 
    if ( name[0] != '\0' && stabstr[cnt] != ':' ) 
```

Comment:
`std::string` isn't guaranteed to be null-terminated. A better check is `!name.empty()`. I know you didn't write this, but we may as well clean it up while we're here.